EST.MIT.InvoiceImporter.Function/Functions/ImporterFunctions.cs

```diff
@@ -21,30 +20,44 @@ public class ImporterFunctions : IImporterFunctions
     public ImporterFunctions(IAzureBlobService azureBlobService, IAzureTableService azureTableService, IEventQueueService eventQueueService, INotificationService notificationService)
     {
         _blobService = azureBlobService;
-        _blobServiceClient = azureBlobService.GetBlobServiceClient();
         _azureTableService = azureTableService;
         _eventQueueService = eventQueueService;
         _notificationService = notificationService;
     }
 
     [FunctionName("MainTrigger")]
     public async Task QueueTrigger(
-        [QueueTrigger("rpa-mit-invoice-importer", Connection = "QueueConnectionString")] string importMessage,
-        IBinder blobBinder,
-        ILogger log)
+            [QueueTrigger("rpa-mit-invoice-importer", Connection = "QueueConnectionString")] string importMessage,
+            IBinder blobBinder,
+            ILogger log)
     {
-        log.LogInformation($"[MainTrigger] Recieved message: {importMessage} at {DateTime.UtcNow.ToLongTimeString()}");
-        using (await _blobService.ReadBLOBIntoStream(importMessage, blobBinder))
-        {
-            var isMoved = await _blobService.MoveFileToArchive(_blobService.GetFileName(), _blobServiceClient);
+        log.LogInformation($"[MainTrigger] Received message: {importMessage} at {DateTime.UtcNow.ToLongTimeString()}");
 
-            if (isMoved)
+        try
+        {
+            using (await _blobService.ReadBLOBIntoStream(importMessage, blobBinder))
             {
-                var importRequest = JsonConvert.DeserializeObject<ImportRequest>(importMessage);
-                importRequest.FileName = $"archive/{importRequest.FileName}";
-                var newImportMessage = JsonConvert.SerializeObject(importRequest);
-                await _azureTableService.UpsertImportRequestAsync(importRequest);
+                BlobServiceClient blobServiceClient = _blobService.GetBlobServiceClient();
+                var isMoved = await _blobService.MoveFileToArchive(_blobService.GetFileName(), blobServiceClient);
```

Comment:
I don;t think we need to be passing the blobServiceClient into a method in the AzureBlobService since the service already hold a private reference to the blobClient. MoveFileToArchive could just pick up the existing client

Reply:
Agreed, I plan to further refactor this but the current ticket is only 1 point.